frontend/test/pytest/test_from_plxpr.py

```diff
@@ -0,0 +1,453 @@
+        assert ov1.aval == ov2.aval
+        if not skip_counts and hasattr(ov1, "count"):
+            assert ov1.count == ov2.count, f"{eqn1}, {ov1.count}, {ov2.count}"
+
+
+class TestErorrs:
+    """Test that errors are raised in unsupported situations."""
+
+    def test_dynamic_shots(self):
+        """Test that a NotImplementedError is raised is shots do not match device shots."""
+
+        dev = qml.device("lightning.qubit", wires=2, shots=50)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.sample(wires=0)
+
+        def f():
+            return circuit(shots=1000)
+
+        qml.capture.enable()
+        jaxpr = jax.make_jaxpr(f)()
+        qml.capture.disable()
+
+        with pytest.raises(NotImplementedError):
+            from_plxpr(jaxpr)()
+
+    def test_operator_without_n_wires(self):
+        """Test that a NotImplementedError is raised for an operator without a n_wires parameter."""
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.adjoint(qml.X(0))
+            return qml.expval(qml.Z(0))
+
+        qml.capture.enable()
+        jaxpr = jax.make_jaxpr(circuit)()
+        qml.capture.disable()
+
+        with pytest.raises(NotImplementedError):
+            from_plxpr(jaxpr)()
+
+    def test_observable_without_n_wires(self):
+        """Test that a NotImplementedError is raised for an observable without n_wires."""
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.expval(qml.X(0) + qml.Y(0))
+
+        qml.capture.enable()
+        jaxpr = jax.make_jaxpr(circuit)()
+        qml.capture.disable()
+
+        with pytest.raises(NotImplementedError):
+            from_plxpr(jaxpr)()
+
+    def test_measuring_eigvals_not_supported(self):
+        """Test that a NotImplementedError is raised for converting a measurement
+        specified via eigvals and wires."""
+
+        dev = qml.device("lightning.qubit", wires=2, shots=50)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.measurements.SampleMP(
+                wires=qml.wires.Wires((0, 1)), eigvals=np.array([-1.0, -1.0, 1.0, 1.0])
+            )
+
+        qml.capture.enable()
+        jaxpr = jax.make_jaxpr(circuit)()
+        qml.capture.disable()
+
+        with pytest.raises(NotImplementedError):
+            from_plxpr(jaxpr)()
+
+    def test_measuring_measurement_values(self):
+        """Test that measuring a MeasurementValue raises a NotImplementedError."""
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            return qml.measurements.ExpectationMP(
+                obs=2
+            )  # classical value like will be used for mcms
+
+        qml.capture.enable()
+        jaxpr = jax.make_jaxpr(circuit)()
+        qml.capture.disable()
+
+        with pytest.raises(NotImplementedError):
```

Comment:
Reusing a suggestion from the code itself.
```suggestion
        with pytest.raises(NotImplementedError, match="Only wire-based and observable-based measurements"):
```